pkg/scheduler/util/non_zero.go

```diff
@@ -72,13 +75,10 @@ func GetNonzeroRequestForResource(resource v1.ResourceName, requests *v1.Resourc
 		}
 		return quantity.Value()
 	default:
-		if IsScalarResourceName(resource) {
-			quantity, found := (*requests)[resource]
-			if !found {
-				return 0
-			}
-			return quantity.Value()
+		quantity, found := (*requests)[resource]
```

Comment:
Why do this? Is it to remove code duplication?

In that case, I think it's better to remove the check in `calculateResourceAllocatableRequest`.

Reply:
Part of https://github.com/kubernetes/kubernetes/pull/101466. Already merged.